### ISSUE TITLE
Download and repair all related profiles for one single app

### ISSUE THREAD

**Issue by Buju77** [fastlane/sigh#190]
Hi,

it would be nice to enhance `sigh download` command to download all related profiles for an app (`-a` parameter).

eg. we have 4 profiles for one app which also supports Today widget:

1. `com.company.ios.myapp AdHoc`
2. `com.company.ios.myapp AppStore`
3. `com.company.ios.myapp.widget AdHoc`
4. `com.company.ios.myapp.widget AppStore`

And it gets way more exiting if the app is also supporting Apple Watch:

5. `com.company.ios.myapp.watchapp AdHoc`
6. `com.company.ios.myapp.watchapp AppStore`
7. `com.company.ios.myapp.watchapp.watchkitextension AdHoc`
8. `com.company.ios.myapp.watchapp.watchkitextension AppStore`

So you end up with **8** profiles just for 1 single app. Apple Watch is so great right? ;)

It would be nice if `-a` is using `containsString:` instead of `isEqualToString:` or an extra flag to download all related profiles for given bundle_identifier. 

It would be nice if this is also implemented for `repair` command.

**Comment by KrauseFx** [comment]
Good idea. Wow, another user from Austria? :+1:

**Comment by Buju77** [comment]
yes, from Linz/Pasching ... Austria ftw ;)

**Comment by pjviii** [comment]
This would be great.

Tried my luck with  ```sigh -a "com.company.myapp.*"``` but it did not work... :-)

**Comment by KrauseFx** [comment]
I gave it some more thought, I'd much rather prefer to just be able to pass an array to the `app_identifier`

**Comment by pjviii** [comment]
Yeah, it makes more sense to do it that way.

**Comment by Buju77** [comment]
yes, pass in an array would also solve the issue :)

**Comment by marcelo-schroeder** [comment]
I also would love to see this as my app has a Today extension as well as a companion Apple Watch.
How do we currently solve this issue? (using fastlane) Do we call sigh in the fastfile multiple times (i.e. once for each bundle ID)? By the way, many thanks for this great tool! :-)

**Comment by KrauseFx** [comment]
@marcelo-schroeder yeah, you should use `fastlane` for that. In particular I recommend using [match](https://github.com/fastlane/match) for that. In particular check out [multiple targets](https://github.com/fastlane/match#multiple-targets)

**Comment by marcelo-schroeder** [comment]
@KrauseFx thanks for the reply. I don't want to introduce match into the mix at this stage. I gave sigh a go by calling it multiple times in the Fastfile (one for each bundle ID that requires a provisioning profile) and everything seems to have worked ok. I got it to deploy to TestFlight and the app, Today widget and watch app, all work correctly. Any gotchas I should know about re this approach? By the way, this is running on a Jenkins server on a dedicated Mac Mini. Thanks again.
